hazelcast/src/main/java/com/hazelcast/map/impl/MapKeyLoader.java

```diff
@@ -439,4 +451,14 @@ public String toString() {
     public void onKeyLoad(ExecutionCallback<Boolean> callback) {
         loadFinished.andThen(callback, execService.getExecutor(MAP_LOAD_ALL_KEYS_EXECUTOR));
     }
+
+    public void promoteToLoadedOnMigration() {
+        state.next(State.LOADING);
+        state.next(State.LOADED);
```

Comment:
can't we do just `state.next(State.LOADED)`? Why do we need `state.next(State.LOADING)`?